packages/editor/src/panels/properties/common/NodeEditor.tsx

```diff
@@ -91,10 +97,17 @@ const NodeEditor = ({
         component && hasComponent(entity, component)
           ? () => {
               const entities = SelectionState.getSelectedEntities()
+
+              //clean up the component from the state for these entities
+              if (name) ComponentDropdownStateFunctions.removeComponentFromComponentStateByEntity(entities, name)
```

Comment:
rather than this logic existing here, we can move this to a useEffect callback, and check if the component is no longer on the entity